### DISCUSSION TITLE
3D Spatial Sound - HowlerJS

### DISCUSSION THREAD

**Discussion by TerraPrimal** [4ian/GDevelop#2349] [Enhancement ideas (and their technical discussions)]
## Description
From what I understand, GDevelop uses HowlerJS for sound. HowlerJS supports the use of 3D Spatial Sound which I think would be a great addition to the software.
Currently sounds are essentially "global" in the sense that we can't have a sound coming from an X/Y loc and have it get louder/dim depending on the distance.

An example of this (although in a 3D plane) is https://howlerjs.com/assets/howler.js/examples/3d/

## Solution suggested
Add HowlerJS's 3D Spatial Sound into GDevelop. Sounds could be attached to objects or set at an X & Y Coordinate. If possible, it would be great if we could somehow manually adjust the range of the sound depending on the zoom level, however that might be going a little far at this point.

**Comment by AtomicCrocStudios** [comment]
Just curious if there has been any update on this? I know that Howler was updated to support this feature, but I haven't heard anything since.

Thanks :)

**Comment by 4ian** [comment]
This was done by @arthuro555 :)
